grails-app/controllers/org/broadinstitute/orsp/ConsentGroupController.groovy

```diff
@@ -126,10 +125,19 @@ class ConsentGroupController extends AuthenticatedController {
      * @return Redirect to the project view page
      */
     def breakLink() {
-        def issue = queryService.findByKey(params.projectKey)
-        def links = ConsentCollectionLink.findAllByProjectKeyAndConsentKey(params.projectKey, params.consentKey)
-        deleteCollectionLinks(links)
-        redirect(controller: "project", action: "main", params: [projectKey: params.projectKey, tab: "consent-groups"])
+        try {
+            queryService.updateCollectionLinkStatus(params.consentKey, params.projectKey, params.type == "unlink" ? CollectionLinkStatus.Unlinked.name : CollectionLinkStatus.Rejected.name)
+            List<ConsentCollectionLink> links = queryService.findConsentCollectionLinksByProjectKeyAndConsentKey(params.projectKey, params.consentKey)
+            deleteCollectionLinks(links)
+            response.status = 200
+            render([message: "Links with the following ids: " + links.collect{it.sampleCollectionId}.join(",")  +
+                    " for the specified consent key: " + params.consentKey + " and project key " + params.projectKey + " are been deleted"] as JSON)
```

Comment:
* nit, language: `are been deleted` -> `have been deleted`
* sampleCollectionId can be null so you might have empty list elements in `links.collect{it.sampleCollectionId}.join(",")`